src/pepr/operator/controllers/istio/ambient-waypoint.ts

```diff
@@ -501,3 +504,56 @@ export async function reconcileExistingResources(
     throw error;
   }
 }
+
+// Generate Waypoint for ambient egress
+export function createEgressWaypointGateway(pkgs: Set<string>, generation: number) {
+  // Add annotations from resource
+  const annotations: Record<string, string> = {};
+  for (const pkgId of pkgs) {
+    annotations[`${getSharedAnnotationKey(pkgId)}`] = "user";
+  }
+
+  // Waypoint resource
+  const waypoint: K8sGateway = {
+    metadata: {
+      name: egressWaypointName,
+      namespace: ambientEgressNamespace,
+      annotations,
+      labels: {
+        "uds/package": sharedEgressPkgId,
+        "uds/generation": generation.toString(),
```

Comment:
For future consideration - this generation label causes the pod(s) to update/cycle each time the generation changes. This means that multiple workloads requiring ambient egress could result in the waypoint cycling a few times. I'd say for now this is okay since the update policy will ensure 0 downtime, but may be an optimization for the future:
- debounce updates
- find a way to save generation on this without cycling the pod 🤔